### ISSUE TITLE
Looks like performance of 30901 (Bitcoin raw private key (P2PKH), compressed) is limited by CPU, not GPU

### ISSUE THREAD

**Issue by konard** [hashcat/hashcat#3956]
```bash
hashcat -a 3 -m 30901 13zb1hQbWVsc2S7ZTZnP2G4undNNpdh5so "000000000000000000000000000000000000000000000002000000?h?h?h?h?h?h?h?h?h?h" -d 2
```

When I use this command with  30901 (Bitcoin raw private key (P2PKH), compressed) it looks like it gives me only 8000-9000 kH/s, and if I see at top

![Screenshot_20240224_162517](https://github.com/hashcat/hashcat/assets/1431904/f83b93ae-7d0a-49ea-9f86-ca69aba06386)

Then only 1 CPU core is used by `hashcat`. May it be a limiting factor for the performance? Is there a way to use more CPUs for data exchange with GPU?

Also for some reason OpenCL gives better performance then CUDA, should I create another issue for that?

**Comment by gitb2010** [comment]
The password length is too long.
I am also trying to find a solution.
Have you already come up with a solution

**Comment by PenguinKeeper7** [comment]
This is a known problem with Hashcat detailed here: https://hashcat.net/faq/morework
TL;DR, for masks, Hashcat parallelises based on the first 4 characters and in your command, it can only make use of a single GPU core, so it may be faster to use -S, where the whole candidate is used for parallelisation. Also, Hashcat would suck for the BTC puzzle and applications that use BGLS / Kangaroo would probably be over trillions of times faster

**Comment by gitb2010** [comment]
How to switch to using the BGLS or Kangaroo algorithm in Hashcat?

**Comment by PenguinKeeper7** [comment]
> How to switch to using the BGLS or Kangaroo algorithm in Hashcat?

You can't, as Hashcat doesn't support those algorithms and you will have to use another tool

**Comment by gitb2010** [comment]
Can you recommend some tools for me?

**Comment by gitb2010** [comment]
> > How to switch to using the BGLS or Kangaroo algorithm in Hashcat?
> 
> You can't, as Hashcat doesn't support those algorithms and you will have to use another tool


Can you recommend some tools for me?

**Comment by PenguinKeeper7** [comment]
https://github.com/brichard19/BitCrack if you don't know the public key 
https://github.com/JeanLucPons/Kangaroo if you do know the public key
